### ISSUE TITLE
Detect if the ID is valid with AI

### ISSUE THREAD

**Issue by emanuelconunaemme** [associazione-rousseau/camelot-flutter#17]
We can integrate services like Rekognition, Tensorflow or native flutter AI libraries to filter non-IDs or bad pictures and ask the user to take another picture.

**Comment by aledeniz** [comment]
@emanuelconunaemme with “native flutter” do you mean client side? How big do you forecast the model to be?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by bhack** [comment]
I think It will be quite hard  to collect a large and robust dataset for this.
Just to leave a pinpoint:
https://arxiv.org/abs/1805.02283

**Comment by bhack** [comment]
Isn't better to integrate with SPID directly instead of handling all these id cards in the Rosseau system?

**Comment by bhack** [comment]
Check:
https://www.agid.gov.it/it/agenzia/stampa-e-comunicazione/notizie/2019/06/26/spid-apre-ai-privati-primo-servizio-accessibile-il-sistema-pubblico-didentita

**Comment by bhack** [comment]
This Is the repo for previously mentioned  Docface and Docface++ followup paper https://github.com/seasonSH/DocFace

**Comment by emanuelconunaemme** [comment]
@bhack yes, it's better to integrate spid. Although, this would cover 10% of the user. For the other 90% adopting some commercial solution (like tensorflow or aws rekognito) would filter 10% of the wrong id. I tried and it works pretty well. Of course, it's pretty inefficient in detecting the positives.

**Comment by bhack** [comment]
I still belive that with limited resources Is better to invest on Spid cause AI will not enable you for https://www.camera.it/leg18/126?tab=1&leg=18&idDocumento=1892&sede=&tipo=

**Comment by bhack** [comment]
I repost https://www.agid.gov.it/it/agenzia/stampa-e-comunicazione/notizie/2019/11/21/consultazione-linee-guida-sulla-sottoscrizione-documenti-tramite-spid

**Comment by bhack** [comment]
So IMHO If we can do both It Is ok but if we need to choiche where to invest time probably it Is better to reserve resources for SPID.
